Add `set_json_payload` method to TestRequest

boustrophedon opened the pull request:
Implements #824 - Add a `TestRequest::set_json_payload` convenience method.

boustrophedon committed:
Add `set_json` method to TestRequest

- Takes a type which implements serde::Serialize, serializes it to JSON,
and sets it as the payload. The content-type is also set to JSON.

fafhrd91 commented on src/test.rs:
```diff
@@ -477,6 +478,14 @@ impl TestRequest {
     }
 
+    /// Set request payload serialized to JSON
+    pub fn set_json_payload<T: Serialize>(mut self, data: &T) -> Self {
```
let's name it `set_json`

fafhrd91 reviewed (approved):
could you also add entry to changes

boustrophedon commented:
I updated CHANGES.md, added a line in the comment about setting the Content-Type header, and added a line to the test to assert the content type is application/json.

boustrophedon commented:
The travis build failed only on the beta build with a spurious error:
```
After this operation, 38.1 MB of additional disk space will be used.
WARNING: The following packages cannot be authenticated!
  libssl-dev libssl1.0.0 openssl
E: There were unauthenticated packages and -y was used without --allow-unauthenticated
The command "sudo apt-get install -y openssl libssl-dev libelf-dev libdw-dev cmake gcc binutils-dev libiberty-dev" failed and exited with 100 during .
```

Can you rerun?

fafhrd91 commented:
thanks!
